### DISCUSSION TITLE
Hardhat support in client provider

### DISCUSSION THREAD

**Discussion by ishaansinghal** [wevm/wagmi#407] [Q&A]
I am trying to create a client with a provider for a local Hardhat node
<img width="468" alt="image" src="https://user-images.githubusercontent.com/17390893/166617848-0b46bad6-fc1c-44f4-99fd-fa25288a0a2a.png">

Unfortunately using the JsonRpcProvider does not work since the network attribute seems to be undefined. Any tips?

**Comment by L-TheG** [comment]
It does work.
I am doing this:
```
const ethProvider = new JsonRpcProvider(process.env.RPC_URL, getNetwork(process.env.CHAIN_ID);
const connector = new MetaMaskConnector({chains: [chain.hardhat]});

const client = createClient({
    autoConnect: true,
    provider: ethProvider,
    connectors: [connector],
});
```
RPC_URL is just the hardhat rpc url so http://127.0.0.1:8545/
CHAIN_ID is the hardhat network chain id so 1337

This works for me

**Comment by shixin-guo** [comment]
if we can add this part into Doc , it will be better
